ably/realtime_channel.go

```diff
@@ -63,9 +68,38 @@ func ChannelWithCipher(params CipherParams) ChannelOption {
 	}
 }
 
+func ChannelWithParams(key string, value string) ChannelOption {
+	return func(o *channelOptions) {
+		if o.Params == nil {
+			o.Params = map[string]string{}
+		}
+		o.Params[key] = value
+	}
+}
+
+func ChannelWithRewind(numberOfMessages string) ChannelOption {
```

Comment:
As the spec doesn't define this as part of the public API, this should be unexported or just removed.

Reply:
Actually, it's being used in the dotnet client, and makes it easy for the developer to use it on the go

Reply:
We're trying to protect ourselves from incompatible spec changes down the line. If tomorrow the spec adds an explicit `rewind` ChannelOption which happens to be different to this one, then we'll have a problem. That's why we should stick to the public API as published in the spec.

Reply:
I will remove it for the time being, but I believe we should raise request for this public API in the docs, because `rewind` is just an extension of params and doesn't make sense for a user to pass hardcoded value for the same, rather it should be exposed via some public interface. We already have it in .net and works well for users ( based on the recent ticket raised by one of the customer)

Reply:
Resolved

Reply:
> I believe we should raise request for this public API in the docs

Sure, that sounds like a good idea.